pkg/apis/ceph.rook.io/v1/network.go

```diff
@@ -32,10 +32,19 @@ func (n *NetworkSpec) IsMultus() bool {
 	return n.Provider == NetworkProviderMultus
 }
 
-// IsHost get whether to use host network provider. This method also preserve
-// compatibility with the old HostNetwork field.
+// IsHost can be used to determine if pods should use host networking by
+// checking whether the cephCluster is configured to use the "host" network provider.
+// This method also maintains compatibility with the old hostNetwork setting
 func (n *NetworkSpec) IsHost() bool {
-	return (n.HostNetwork && n.Provider == NetworkProviderDefault) || n.Provider == NetworkProviderHost
+	return (n.HostNetwork && n.Provider == NetworkProviderDefault) || n.Provider == NetworkProviderHost || n.Provider == NetworkProviderHostStrict
+}
+
+// IsHostStrict determines whether the cephCluster is configured to use the use host
+// or networking on all pods. This is achieved by checking whether the "host-strict" network provider
+// is set on the cephCluster.
+// This method does not preserve compatibility with the old HostNetwork setting
```

Comment:
Prefer listing field keys using the lower-camel-case format that users see more often when configuring YAML manifests -- versus the internal representation that usually only Rook sees that has upper-camel-case.

```suggestion
// This method does not preserve compatibility with the legacy hostNetwork setting
```

Reply:
Don't forget this past suggestion as well to help users understand the difference between "host" and "host-strict": https://github.com/rook/rook/pull/13651/files#r1473323308